code/game/objects/items/blueprints.dm

```diff
@@ -63,15 +64,25 @@
 
 /obj/item/areaeditor/blueprints/attack_self(mob/user)
 	. = ..()
-	var/area/A = get_area()
-	if(get_area_type() == AREA_STATION)
-		. += "<p>According to \the [src], you are now in <b>\"[html_encode(A.name)]\"</b>.</p>"
-		. += "<p>You may <a href='?src=\ref[src];edit_area=1'>make an amendment</a> to the drawing.</p>"
-	if(!viewing)
-		. += "<p><a href='?src=\ref[src];view_blueprints=1'>View structural data</a></p>"
+	if(!legend)
+		var/area/A = get_area()
+		if(get_area_type() == AREA_STATION)
+			. += "<p>According to \the [src], you are now in <b>\"[html_encode(A.name)]\"</b>.</p>"
+			. += "<p>You may <a href='?src=\ref[src];edit_area=1'>make an amendment</a> to the drawing.</p>"
+		. += "<p><a href='?src=\ref[src];view_legend=1'>View wire colour legend</a></p>"
+		if(!viewing)
+			. += "<p><a href='?src=\ref[src];view_blueprints=1'>View structural data</a></p>"
+		else
+			. += "<p><a href='?src=\ref[src];refresh=1'>Refresh structural data</a></p>"
+			. += "<p><a href='?src=\ref[src];hide_blueprints=1'>Hide structural data</a></p>"
 	else
-		. += "<p><a href='?src=\ref[src];refresh=1'>Refresh structural data</a></p>"
-		. += "<p><a href='?src=\ref[src];hide_blueprints=1'>Hide structural data</a></p>"
+		if(legend == 1)
```

Comment:
This check is specific because if it's not exactly equal to 1/0 (I'll change it to TRUE/FALSE) then it's actually on the second level of the legend (See L67 and L83).